src/main/java/seedu/address/model/tut/Tutorial.java

```diff
@@ -11,182 +11,402 @@
+     * @throws NoTutorialException if called on a None instance.
+     */
+    public abstract void markAttendance(Student student, TutDate tutDate);
+
+    /**
+     * Checks if a student is in the list of students.
+     *
+     * @param student The student to check.
+     * @return True if the student is in the list; otherwise false.
+     */
+    public abstract boolean studentInList(Student student);
+
+    /**
+     * Deletes a student from the tutorial.
+     *
+     * @param student The student to be removed.
+     */
+    public abstract void deleteStudent(Student student);
+
+    private static Tutorial exist(TutName tutName, TutorialClass tutorialClass) {
+        return new Exist(tutName, tutorialClass);
     }
 
-    public List<TutDate> getTutDates() {
-        return new ArrayList<>(tutDates.values());
+    /**
+     * Creates a tutorial that represents no tutorial.
+     *
+     * @return A None tutorial object.
+     */
+    public static Tutorial none() {
+        return None.none();
     }
 
-    public TutorialClass getTutorialClass() {
-        return tutorialClass;
+    /**
+     * Creates an instance of an existing tutorial.
+     *
+     * @param tutName The name of the tutorial.
+     * @param tutorialClass The class of the tutorial.
+     * @return A new tutorial object.
+     */
+    public static Tutorial of(TutName tutName, TutorialClass tutorialClass) {
+        requireNonNull(tutName);
+        requireNonNull(tutorialClass);
+        return Tutorial.exist(tutName, tutorialClass);
     }
 
     /**
-     * Marks the attendance for the student for the particular tutorial date.
+     * Represents a None type of tutorial (non-existent).
      */
-    public void markAttendance(Student student, TutDate tutorialDate) throws TutDateNotFoundException {
-        if (!isValidTutorialDate(tutorialDate)) {
-            throw new TutDateNotFoundException();
+    private static final class None extends Tutorial {
+        private static final None none = new None();
+
+        private final List<Student> students = new ArrayList<>();
+
+        public static None none() {
+            return none;
         }
-        if (!studentInList(student)) {
-            add(student);
+
+        @Override
+        public void add(Student student) {
+            if (student != null && !students.contains(student)) {
+                students.add(student);
+            }
+        }
+
+        @Override
+        public List<Student> getStudents() {
+            return this.students;
+        }
+
+        @Override
+        public boolean setAttendance(Date date, StudentId target) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public void assignStudent(Student student) {
+            requireNonNull(student);
+            students.add(student);
+        }
+
+        @Override
+        public TutName getTutName() {
+            throw new NoTutorialException();
+        }
+
```

Comment:
I'm not sure about the original intention of having get student method here, but I wonder should getting student from the name here be considered invalid? Like should it not throw exception and return the student instead? It doesn't seem to be violating the rules by returning the student.